### PR TITLE
OS-8108 Support vnc password for BHYVE

### PR DESCRIPTION
**Pull request by jasonbking**

### CODE REVIEW

File: usr/src/cmd/bhyve/Makefile [hunk 1]
```diff
@@ -150,8 +155,6 @@ clean: $(SUBDIRS)
 clobber: clean $(SUBDIRS)
 	$(RM) $(CLOBBERFILES)
 
-lint:	lint_SRCS $(SUBDIRS)
```
Comment (bahamat) [inline usr/src/cmd/bhyve/Makefile]: Why is this removed?

File: usr/src/cmd/bhyve/rfb.c [hunk 2]
```diff
@@ -803,7 +803,7 @@ rfb_handle(struct rfb_softc *rc, int cfd)
 		 * The client then sends the resulting 16-bytes response.
 		 */
 #ifndef NO_OPENSSL
-		strncpy(keystr, rc->password, PASSWD_LENGTH);
+		strncpy((char *)keystr, rc->password, PASSWD_LENGTH);
```
Comment (bahamat) [inline usr/src/cmd/bhyve/rfb.c]: Does the copyright in this file need updating?

File: usr/src/cmd/bhyve/Makefile (hunk 1 above)
Comment (jasonbking) [reply usr/src/cmd/bhyve/Makefile]: We've stopped using lint some time ago and use smatch in it's place now. Since it'd be a massive change to go through and remove all the now unused targets, the convention has been to remove them as you touch the makefiles.  Smatch is run as a shadow compiler, so it doesn't need any special makefile rules
